File: chrome/browser/resources/chromeos/zip_archiver/unpacker-test/js/test_utils.js
// Copyright 2015 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

'use strict';

/**
 * Utility functions for all tests.
 */
var testUtils = {
  /**
   * Forces failure in tests. Should be called only from 'beforeEach',
   * 'afterEach' and 'it'. Useful to force failures in promises.
   * @param {Object|string} An error with stack trace or a string error that
   *     describes the failure reason.
   */
  forceFailure: function(error) {
    console.error(error.stack || error);
    setTimeout(function() {
      expect(false).to.be.true;
    });
  }
};

File: chrome/renderer/safe_browsing/test_utils.cc
// Copyright (c) 2012 The Chromium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/renderer/safe_browsing/test_utils.h"

#include <map>
#include <string>

#include "chrome/renderer/safe_browsing/features.h"
#include "testing/gmock/include/gmock/gmock.h"

namespace safe_browsing {

void ExpectFeatureMapsAreEqual(const FeatureMap& first,
                               const FeatureMap& second) {
  std::map<std::string, double> sorted_first(first.features().begin(),
                                             first.features().end());
  std::map<std::string, double> sorted_second(second.features().begin(),
                                              second.features().end());
  EXPECT_THAT(sorted_first, testing::ContainerEq(sorted_second));
}

}  // namespace safe_browsing
